internal/semver/semver.go

```diff
@@ -0,0 +1,70 @@
+package semver
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/hashicorp/go-version"
+	"github.com/hashicorp/vault/api"
+
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+)
+
+func getTargetVaultVersion(client *api.Client) (string, error) {
```

Comment:
It would be great if this took a context and used the context-aware vault client functions, now that they're available.

Reply:
Great idea! I refactored the code to use the context-aware functions